### PR TITLE
Groups as Board Moderators

### PR DESCRIPTION
**Pull request by Oldiesmann**
These changes allow you to assign a membergroup to moderate a board.

### CODE REVIEW

File: Sources/ManageBoards.php [hunk 1]
```diff
@@ -548,6 +548,26 @@ function EditBoard()
...
 		list ($context['board']['last_moderator_id']) = array_slice(array_keys($context['board']['moderators']), -1);
+		
+	// Get groups that can moderate this board
+	// We already have the group names loaded, so this is easy
+	$request = $smcFunc['db_query']('', '
+		SELECT id_group
+		FROM {db_prefix}moderator_groups
+		WHERE id_board = {int:current_board}',
+		array(
+			'current_board' => $_REQUEST['boardid'],
```
Comment (ghost) [inline Sources/ManageBoards.php:559]: Might be nice to see an (int) typecast here, just cause...

**Comment by ghost** [comment]
Well done. This commit is Craw approved. :P

File: Sources/ManageBoards.php (hunk 1 above)
Comment (Oldiesmann) [inline Sources/ManageBoards.php:559]: SMF already does that earlier on in the function:

```
// id_board must be a number....
$_REQUEST['boardid'] = isset($_REQUEST['boardid']) ? (int) $_REQUEST['boardid'] : 0;
```

**Comment by Oldiesmann** [comment]
Just found out that the development branch is older than the release branch, so I'm going to kill this pull request and submit a new one based on the release branch instead.

**Comment by emanuele45** [comment]
Remember that before mixing the branches there is to decide what to do with the other PRs pending around.
They have been around for quite a long time and ask the authors to re-align to the new code would be pretty rude...

There is a topic about that somewhere on a team board.

File: Sources/ManageBoards.php (hunk 1 above)
Comment (ghost) [inline Sources/ManageBoards.php:559]: hey
[url=http://halftruth2014.com]bro half truth[/url]

**Comment by Oldiesmann** [comment]
I didn't realize that the development branch was outdated or I would have submitted this using the release branch instead, which I'm working on doing now.

**Pull request closed without merging**